Configuration/TCA/Overrides/tx_tea_domain_model_product_tea.php

```diff
@@ -0,0 +1,26 @@
+<?php
+
+defined('TYPO3') || die();
+
+$typo3Version = new \TYPO3\CMS\Core\Information\Typo3Version();
+
+if ($typo3Version->getMajorVersion() < 12) {
```

Comment:
I propose moving this from the overrides directly to `Configuration/TCA/tx_tea_domain_model_product_tea.php`.

Reply:
The files contained directly TCA folder should contain only a direct return of the TCA array. News is for example also solving it like that: https://github.com/georgringer/news/blob/main/Configuration/TCA/Overrides/tx_news_domain_model_news.php

Reply:
> The files contained directly TCA folder should contain only a direct return of the TCA array.

Then let's do that. Or am I missing something here?

Usually, for version-specific switches, I do it like that in my extensions:

```php
$tca = […];
// modify $tca depending on which TYPO3 version we're on
return $tca;
```

Reply:
I think as long as you don't do anything that mustn't be cached that sounds fine - and versions shouldn't change within the cache lifetime